compilers/concrete-compiler/compiler/lib/Dialect/RT/Analysis/LowerDataflowTasksToRT.cpp

```diff
@@ -591,6 +595,23 @@ struct FinalizeTaskCreationPass
         op->setOperand(1, clone);
       }
     });
+
+    module.walk([&](RT::WorkFunctionReturnOp op) {
```

Comment:
Not related to the PR (should you exclude the first commit)? If included why we always copy results buffers?

Reply:
This may indeed be better suited for inclusion in #824 if @antoniupop confirms that the patch should be included.

Reply:
This is actually a bugfix - the copy is needed when the return from a task is allocated outside the task as we can't control when deallocation occurs, so the task return may have been deallocated before its use otherwise.
This commit contains fixes needed by the PR, but is not quite ready for merging. Could be split into separate PR.

Reply:
Comments are weirdly out of order - @andidr parts of the commit are needed for #824 as otherwise all relevant inputs will crash the compiler.

Reply:
I will move 129c3c42151eb705ad867a5eb221c753e8d9d28a to #824.